verify.sh

```
@@ -0,0 +1,69 @@
+command -v cmake >/dev/null 2>&1 || { echo -e "${RED}Error: cmake is required but not installed.${NC}"; exit 1; }
+command -v go >/dev/null 2>&1 || { echo -e "${RED}Error: go is required but not installed.${NC}"; exit 1; }
+command -v make >/dev/null 2>&1 || { echo -e "${RED}Error: make is required but not installed.${NC}"; exit 1; }
+echo "All tools found."
+
+# 2. Build C++ Core
+echo -e "\n${GREEN}[Step 2] Building C++ Storage Engine (LSM Core)...${NC}"
+cd "$PROJECT_ROOT/cpp/lsm"
+mkdir -p build
+cd build
+cmake ..
+make -j$(nproc)
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)